src/components/ReviewComponents/submit-utils.ts

```diff
@@ -0,0 +1,110 @@
+import { Dispatch } from 'redux';
+import { addNotification } from '@redhat-cloud-services/frontend-components-notifications/redux';
+import { createApplication, createComponent } from '../../utils/create-utils';
+import { FormState } from '../form-context';
+import { ReviewComponentsFormValues } from './types';
+import { transformResources } from './utils';
+
+const createComponents = async (
+  components: ReviewComponentsFormValues['components'],
+  application: string,
+  namespace: string,
+  dryRun?: boolean,
+) => {
+  return Promise.all(
+    Object.values(components).map((component) => {
+      return createComponent(
+        {
+          name: component.name,
+          gitRepo: component.source,
+          replicas: component.replicas && Number(component.replicas),
+          targetPort: component.targetPort && Number(component.targetPort),
+          resources: component.resources && transformResources(component.resources),
+        },
+        application,
+        namespace,
+        dryRun,
+      );
+    }),
+  );
+};
+
+export const createResources = async (
+  formState: FormState,
+  components: ReviewComponentsFormValues['components'],
+  dispatch: Dispatch,
+) => {
+  const shouldCreateApplication = !formState.existingApplication;
+  let appName = formState.existingApplication;
+  if (shouldCreateApplication) {
+    try {
+      const appData = await createApplication(formState.application, formState.namespace, true);
+      appName = appData.metadata.name;
+    } catch (error) {
+      dispatch(
+        addNotification({
+          variant: 'danger',
+          title: 'Application creation failed!!',
+          description: error.message,
+        }),
+      );
+      throw error;
+    }
+  }
+
+  try {
+    await createComponents(components, appName, formState.namespace, true);
```

Comment:
Shouldn't this only create component when application creation is successful?

Reply:
Yes, I'm rethrowing the error in the catch block above, so this function should not continue if the above call fails.